web/src/popover_menus.js

```diff
@@ -272,6 +288,140 @@ export function initialize() {
+
+                const topic = $(e.currentTarget).attr("data-topic-name");
+                muted_topics_ui.mute_topic(stream_id, topic);
+                e.stopPropagation();
+                e.preventDefault();
+                instance.hide();
+            });
+
+            $popper.one("click", ".sidebar-popover-unmute-topic", (e) => {
+                const stream_id = topic_popover_stream_id(e);
+                if (!stream_id) {
+                    return;
+                }
+
+                const topic = $(e.currentTarget).attr("data-topic-name");
+                muted_topics_ui.unmute_topic(stream_id, topic);
+                e.stopPropagation();
+                e.preventDefault();
+                instance.hide();
+            });
+
+            $popper.one("click", ".sidebar-popover-mark-topic-read", (e) => {
+                const stream_id = topic_popover_stream_id(e);
+                if (!stream_id) {
+                    return;
+                }
+
+                const topic = $(e.currentTarget).attr("data-topic-name");
+                unread_ops.mark_topic_as_read(stream_id, topic);
+                e.stopPropagation();
+                instance.hide();
+            });
+
+            $popper.one("click", ".sidebar-popover-delete-topic-messages", (e) => {
+                const stream_id = topic_popover_stream_id(e);
+                if (!stream_id) {
+                    return;
+                }
+
+                const topic = $(e.currentTarget).attr("data-topic-name");
+                const args = {
+                    topic_name: topic,
+                };
+
+                const html_body = render_delete_topic_modal(args);
+
+                confirm_dialog.launch({
+                    html_heading: $t_html({defaultMessage: "Delete topic"}),
+                    help_link: "/help/delete-a-topic",
+                    html_body,
+                    on_click() {
+                        message_edit.delete_topic(stream_id, topic);
+                    },
+                });
+
+                e.stopPropagation();
+                instance.hide();
+            });
+
+            $popper.one("click", ".sidebar-popover-toggle-resolved", (e) => {
+                const $topic_row = $(e.currentTarget);
+                const stream_id = Number.parseInt($topic_row.attr("data-stream-id"), 10);
+                const topic_name = $topic_row.attr("data-topic-name");
+                message_edit.with_first_message_id(stream_id, topic_name, (message_id) => {
+                    message_edit.toggle_resolve_topic(message_id, topic_name);
+                });
+
+                e.stopPropagation();
+                e.preventDefault();
+                instance.hide();
+            });
+
+            $popper.one("click", ".sidebar-popover-move-topic-messages", (e) => {
+                const $topic_row = $(e.currentTarget);
+                const stream_id = Number.parseInt($topic_row.attr("data-stream-id"), 10);
```

Comment:
If there any particular reason why we are not using the `topic_popover_stream_id` function in this case?